report_viewer/apps/viewer/middleware/report_views_middleware.py

```diff
@@ -98,22 +94,25 @@ def fingerprint_codename(self, fingerprint_hash: int) -> str:
 
     def __call__(self, request):
         try:
-            absolute_uri: str = request.build_absolute_uri()
-            guid_index = -1
-            if absolute_uri[-1] == "/":
-                guid_index = -2
-
-            guid: str = absolute_uri.split("/")[guid_index]
-            if len(guid) == 36 and UUID(guid, version=4):
-                string_to_hash = self.user_fingerprint(request, SECRET_KEY)
-                fingerprint_hash = self.four_digit_hash(string_to_hash)
-                fingerprint_codename = self.fingerprint_codename(fingerprint_hash)
-                ReportVisitsMetrics(
-                    case=S3Report.objects.get(guid=guid).case,
-                    guid=guid,
-                    fingerprint_hash=fingerprint_hash,
-                    fingerprint_codename=fingerprint_codename,
-                ).save()
+            string_to_hash: str = self.user_fingerprint(request)
+            fingerprint_hash: int = self.four_digit_hash(string_to_hash)
+            if not UserCacheUniqueHash.objects.filter(
```

Comment:
If all you wanted to do is exclude users logged in to the platform, just checking `request.user.is_authenticated == False` here would be sufficient.

Reply:
Just tested this isn't possible as it will be across two different domains